File: 1668.maximum-repeating-substring.go
// @leet start
func maxRepeating(sequence string, word string) int {
  max := 0
  for i := 0; i < len(sequence); i++ {
    if sequence[i] != word[0] { continue }
    if len(sequence[i:]) < len(word) { break }
    count := 0
    j := i
    for sequence[j:j+len(word)] == word {
      count++
      if count > max { max = count }
      j += len(word)
      if len(sequence[j:]) < len(word) { break }
    }
  }
  return max
}
// @leet end

File: 1668.maximum-repeating-substring.ts
// @leet start
function maxRepeating(sequence: string, word: string): number {
  let max = 0;
  for (let i = 0; i < sequence.length; i++) {
    let count = 0;
    let j = i;
    while (sequence.substring(j).indexOf(word) === 0) {
      count++;
      if (j + word.length > sequence.length) break;
      j += word.length;
    }
    if (count > max) max = count;
  }
  return max;
};
// @leet end
